lib/_http_incoming.js

```diff
@@ -119,6 +119,146 @@ function _addHeaderLines(headers, n) {
 }
 
 
+// This function is used to help avoid the lowercasing of a field name if it
+// matches a 'traditional cased' version of a field name. It then returns the
+// lowercased name to both avoid calling toLowerCase() a second time and to
+// indicate whether the field was a 'no duplicates' field. If a field is not a
+// 'no duplicates' field, a `0` byte is prepended as a flag. The one exception
+// to this is the Set-Cookie header which is indicated by a `1` byte flag, since
+// it is an 'array' field and thus is treated differently in _addHeaderLines().
+// TODO: perhaps http_parser could be returning both raw and lowercased versions
+// of known header names to avoid us having to call toLowerCase() for those
+// headers.
+/* eslint-disable max-len */
+// 'array' header list is taken from:
+// https://mxr.mozilla.org/mozilla/source/netwerk/protocol/http/src/nsHttpHeaderArray.cpp
+/* eslint-enable max-len */
+function matchKnownFields(field) {
+  var low = false;
+  while (true) {
+    switch (field) {
+      case 'Content-Type':
```

Comment:
I had to stop somewhere, plus you can only have a switch statement so large before the function gets deopt'ed (I *think* this may still be the case anyway).